pkg/yang/types_builtin.go

```diff
@@ -794,3 +930,17 @@ func (r YangRange) Contains(s YangRange) bool {
 	}
 	return true
 }
+
+// Frac returns the fractional part of f.
+func Frac(f float64) float64 {
+	return f - math.Trunc(f)
+}
+
+// pow10 returns 10^e without checking for overflow.
+func pow10(e uint8) uint64 {
+	var out uint64 = 1
+	for i := uint8(0); i < e; i++ {
+		out *= 10
+	} 
+	return out
+}
```

Comment:
var pow10 = []uint64{
1,
10,
100,
1000,
10000,
100000,
1000000,
10000000,
100000000,
1000000000,
10000000000,
100000000000,
1000000000000,
10000000000000,
100000000000000,
1000000000000000,
10000000000000000,
100000000000000000,
1000000000000000000,
}

And you can saw pow10[n.FractionDigits]

(You don't have to do this, but I though O(1) was kinda fun :-)